test-utils/src/lib.rs

```diff
@@ -45,3 +50,139 @@ pub fn remove_job(role_type: RoleType, job_id: JobId) {
     SubmittedJobs::<Runtime>::remove(role_type, job_id);
     SubmittedJobsRole::<Runtime>::remove(job_id);
 }
+
+pub trait ProtocolTestingUtils: FullProtocolConfig
+where
+    <<Self as FullProtocolConfig>::Client as ProvideRuntimeApi<
+        <Self as FullProtocolConfig>::Block,
+    >>::Api: JobsApiForGadget<<Self as FullProtocolConfig>::Block>,
+{
+    fn setup_test_node(
+        additional_param: <Self as FullProtocolConfig>::AdditionalTestParameters,
+        node_input: NodeInput,
+    ) -> impl SendFuture<'static, ()>;
+}
+
+#[macro_export]
+macro_rules! generate_tss_tests {
+    ($config:ident, $threshold_sig_ty:expr) => {
+        #[cfg(test)]
+        mod tests {
+            use std::sync::Arc;
+            use tangle_primitives::jobs::{
+                DKGTSSPhaseOneJobType, DKGTSSPhaseTwoJobType, JobId, JobSubmission, JobType,
+            };
+            use tangle_primitives::roles::{RoleType, ThresholdSignatureRoleType};
+            use test_utils::mock::{id_to_public, Jobs, MockBackend, RuntimeOrigin};
+            use test_utils::sync::substrate_test_channel::MultiThreadedTestExternalities;
+
+            #[tokio::test(flavor = "multi_thread")]
+            async fn test_externalities_gadget_starts() {
+                test_utils::setup_log();
+                new_test_ext::<1>()
+                    .await
+                    .execute_with_async(|| {
+                        assert_eq!(1, 1);
+                    })
+                    .await
+            }
+
+            #[tokio::test(flavor = "multi_thread")]
+            async fn test_externalities_keygen() {
+                test_utils::setup_log();
+                const N: usize = 3;
```

Comment:
These values should become part of the macro. We need to be able to test and even consider benchmarking under diff settings.